commit d7bb801f703d6a03ea2c6e4b546f40f961ac6691
Author: Bernhard Rosenkränzer <bero@lindev.ch>
Date:   Sat Jun 7 15:09:59 2025 +0200

    Fix LLVM_SPIRV_BACKEND_TARGET_PRESENT detection when building inside the LLVM tree
    
    Signed-off-by: Bernhard Rosenkränzer <bero@lindev.ch>

diff --git a/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt b/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt
index f9f9de4c..3a0e14aa 100644
--- a/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt
+++ b/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt
@@ -102,13 +102,6 @@ if(LLVM_SPIRV_BUILD_EXTERNAL)
 
   message(STATUS "Found LLVM: ${LLVM_VERSION}")
 
-  is_llvm_target_library("SPIRV" spirv_present_result INCLUDED_TARGETS)
-  if(spirv_present_result)
-    message(STATUS "Found SPIR-V Backend")
-    set(SPIRV_BACKEND_FOUND TRUE)
-    add_compile_definitions(LLVM_SPIRV_BACKEND_TARGET_PRESENT)
-  endif()
-
   option(CCACHE_ALLOWED "allow use of ccache" TRUE)
   find_program(CCACHE_EXE_FOUND ccache)
   if(CCACHE_EXE_FOUND AND CCACHE_ALLOWED)
@@ -118,6 +111,13 @@ if(LLVM_SPIRV_BUILD_EXTERNAL)
   endif()
 endif()
 
+is_llvm_target_library("SPIRV" spirv_present_result INCLUDED_TARGETS)
+if(spirv_present_result)
+  message(STATUS "Found SPIR-V Backend")
+  set(SPIRV_BACKEND_FOUND TRUE)
+  add_compile_definitions(LLVM_SPIRV_BACKEND_TARGET_PRESENT)
+endif()
+
 set(LLVM_SPIRV_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 # first try locating SPIRV-Tools via pkgconfig (the old way)
